drivers/hwinfo/hwinfo_mcux_syscon.c

```diff
@@ -22,7 +27,15 @@ ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length)
 		length = DT_INST_REG_SIZE(0);
 	}
 
-	memcpy(buffer, uid_addr, length);
+	if (length > sizeof(dev_id.id)) {
```

Comment:
Why two checks?

Reply:
I originally thought `nxp,lpc-uid` binding could support larger than 128b UUID, in which case it would exceed the size of `struct uid` and I wanted to ensure no writing outside the space reserved for `id`.

On second look it seems `nxp,lpc-uid` is specifically a 128b UUID. I'll remove this secondary length check and add a comment for `id` member of `struct uid`.

Reply:
I would rather define the array size from the device tree reg size property.

Reply:
Updated with a macro using `DT_INST_REG_SIZE(0)` for the number of UUID words.